pkg/types/field.go

```diff
@@ -401,7 +401,7 @@ func (f *Field) AddToResource(g *Builder, r *resource, typeNames *TypeNames, add
 // an earlier step, so they cannot be included as well. Plus probably they
 // should also not change for Create and Update steps.
 func (f *Field) isInit() bool {
-	return !f.Identifier && f.Reference == nil && (f.TFTag != "-" || f.Injected)
+	return !f.Identifier && f.TFTag != "-"
```

Comment:
Please see the comment above. Does the following work?
```suggestion
	return !f.Identifier && (f.TFTag != "-" || f.Injected)
```

Reply:
Thanks @ulucinar for catching this. After rebasing, I tried to solve the conflicts and it seems this removal is not an intentional change. I put this again.
Done!